assert/assertions.go

```diff
@@ -504,8 +504,8 @@ func formatUnequalValues(expected, actual interface{}) (e string, a string) {
 			fmt.Sprintf("%T(%s)", actual, truncatingFormat(actual))
 	}
 	switch expected.(type) {
-	case time.Duration:
-		return fmt.Sprintf("%v", expected), fmt.Sprintf("%v", actual)
+	case time.Duration, uint, uint8, uint16, uint32, uint64:
+		return fmt.Sprint(expected), fmt.Sprint(actual)
```

Comment:
Formatting these with "%[1]T(%[1]v)" would be better so that we don't say two apparently identical numbers are different. Otherwise we regress `assert.Equal(t, int(1), uint(1))`.

Reply:
Also, using this format for all numeric types would fix #912, but this is creeping somewhat, perhaps better in a follow up PR.